processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingOptions.java

```diff
@@ -207,6 +207,9 @@ private static boolean isConsistent(MappingGem gem, ExecutableElement method,
         if ( gem.source().hasValue() && gem.constant().hasValue() ) {
             message = Message.PROPERTYMAPPING_SOURCE_AND_CONSTANT_BOTH_DEFINED;
         }
+        else if (gem.expression().hasValue() && gem.conditionQualifiedByName().hasValue()) {
```

Comment:
formatting nitpick: please add a whitespace (see the formatter in [CONTRIBUTING.md](https://github.com/mapstruct/mapstruct/blob/main/CONTRIBUTING.md)) - 6 lines beyond there is also a whitespace missing, would be nice if you could also add it there 

(wondering why checkstyle did not fail)